### DISCUSSION TITLE
MSP2 Message Problem

### DISCUSSION THREAD

**Discussion by willishf** [iNavFlight/inav#8971] [Q&A]
In the process of implementing java based control software that would run on google coral/jetson nano/pi attached to drone and send MSP commands to FC for autonomous flight. Had no problem implementing code for function id < 255 for the V1 messages. 

Was testing MSPV2_INAV_ANALOG and do not get a response back from FC. 

I have verified that I am generating the correct byte array for the message, in particular the checksum.  Used the "Hello World" message at https://github.com/iNavFlight/inav/wiki/MSP-V2 to verify correctness. 

For MSPV2_INAV_ANALOG would send the following. 
bytearray($X<\x00\x02\x20\x00\x00\xB8)

Thought it might require longer time for response if sampling X number of analog ports but that doesn't seem to be the issue. 

Then tested a couple other V2 commands and no response. 

Assuming I should be requesting data as the master.

Am I missing anything obvious?

Running V6 and INAV Config works as expected.

**Comment by stronnag** [comment]
You have a fault in your code. MSPv2 works fine for compliant implementations.

**Comment by DzikuVx** [comment]
I confirm that MSPv2 is working like expected

**Comment by stronnag** [comment]
As can numerous public examples, inter alia:

* [mwptools](https://github.com/stronnag/mwptools)
* [bbl2kml](https://github.com/stronnag/bbl2kml)
* [inav-follow-me](https://github.com/stronnag/inav-follow-me)
* [impload](https://github.com/stronnag/impload)
* [msp-set-rx](https://github.com/stronnag/msp_set_rx)
* [msp-rs](https://github.com/stronnag/msp-rs)
* [msp-go](https://github.com/stronnag/msp-go)
* [INAV Configurator](https://github.com/iNavFlight/inav-configurator)

**Comment by willishf** [comment]
Thanks for checking. Wanted to mainly make sure I wasn't missing something
obvious.

Same function call to send < 255 functions but build v2 byte array in
another method.

Stumped as to why it isn't responding. Get back 0 bytes for the read.

**Reply by netIOTil** [reply]
did you mange to a get response?

**Reply by sensei-hacker** [reply]
If you sniff the serial port to compare what you are sending vs what Configurator sends, that will generally show exactly what the problem is.

**Comment by willishf** [comment]
Don't recall. No longer working on that project.
